src/mode/modeHandlerMap.ts

```diff
@@ -8,7 +8,16 @@ class ModeHandlerMapImpl {
     let modeHandler = this.modeHandlerMap[key];
     if (!modeHandler) {
       isNew = true;
-      modeHandler = await new ModeHandler();
+      await new Promise((res, rej) => {
```

Comment:
Why was this necessary?

Reply:
`ModeHandler` had async calls in its constructor that were not handled. I added the callback as an _initializer_ of sorts. I'm not a big fan of this method. We could add a static function and make the constructor private?